packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/MethodCallHandlerImpl.java

```diff
@@ -49,8 +49,9 @@ final class MethodCallHandlerImpl implements MethodChannel.MethodCallHandler {
     this.permissionsRegistry = permissionsAdder;
     this.textureRegistry = textureRegistry;
 
-    methodChannel = new MethodChannel(messenger, "plugins.flutter.io/camera");
-    imageStreamChannel = new EventChannel(messenger, "plugins.flutter.io/camera/imageStream");
+    methodChannel = new MethodChannel(messenger, "plugins.flutter.io/camera_android");
```

Comment:
is it part of the effort that move logic from platform to dart?

Reply:
Renaming the method channels is to prevent a mistake I made in one early case where I didn't actually set up Dart registration correctly, but didn't notice because the platform-specific and shared method channels were still identical, so everything (temporarily) worked. That would have cause problems later as the implementations started to diverge, but it would have been very hard to debug. This causes any such mistake to be obvious immediately.

It's also proved useful for some edge cases. E.g., debugging issues with people running plugins in isolates, where there's no Dart auto-registration. This means they fail with an error message that we can easily see a root cause for, rather than having subtle breakage like a mismatch in arguments as channel implementations start to diverge.

Reply:
Gotcha. Are we moving towards a world where ideally the platform native code simply one-line adapts native API, so channel method names and params likely diverge across multiple platforms in that world?

Reply:
That's a direction we're exploring (currently with webview), but it's not the only outcome where this is useful. The design doc linked from the associated bug has a discussion of the the benefits here.

Recent examples of places where just having flexibility helped us include:
- https://github.com/flutter/plugins/pull/5448#discussion_r872754728 - not needing to send a bunch of pointless information over the channel only to do extra work to extract just the relevant part on the native side.
- https://github.com/flutter/plugins/pull/5285 - Converting the iOS image picker implementation to Pigeon, fixing a fragile untyped response object that had caused crashes in a recent change, without having to also rewrite Android at the same time (which would have been significantly more difficult).